src/main/java/se/kth/ContinuousIntegrationServer.java

```diff
@@ -61,16 +65,52 @@ public void handle(String target,
      */
     static void cloneRepository(String url, String branch, String directory) throws InterruptedException, IOException {
         try {
-            // just delete the file directory in case it exists for a clean git clone, this is easier than git pull
+            // just delete the file directory in case it exists for a clean git clone, this
+            // is easier than git pull
             FileUtils.deleteDirectory(new File(directory));
         } catch (Throwable _) {
             // ignore if we can even delete it or not
         }
 
-        // spawn the process for git cloning and wait, this is easier than importing a lib
+        // spawn the process for git cloning and wait, this is easier than importing a
+        // lib
         Process process = new ProcessBuilder("git", "clone", url, "-b", branch, directory).start();
         if (process.waitFor() != 0) {
-            throw new IOException("Bad exitcode (" + process.exitValue() + ") in git clone due to: " + new String(process.getErrorStream().readAllBytes()));
+            throw new IOException("Bad exitcode (" + process.exitValue() + ") in git clone due to: "
+                    + new String(process.getErrorStream().readAllBytes()));
+        }
+    }
+
+    /**
+     * Compiles the cloned repository
+     *
+     * @param file      the cloned file to compile
+     */
+    static void compileRepository() throws InterruptedException, IOException {
+        // spawn the process for compiling (and running?) and wait
+        Process process = new ProcessBuilder("mvn", "compile")
+            .directory(new File(buildDirectory)) // same as  "cd .serverbuild mvn compile" have to go to the correct dir
+            .start();
+        if (process.waitFor() != 0) {
```

Comment:
I expect this pattern to be common when using ProcessBuilder, so maybe make it a function.